### ISSUE TITLE
Read-only variables are editable if you insert text at position 0

### ISSUE THREAD

**Issue by dselman** [accordproject/web-components#212]
# Bug Report 🐛
Variables that are marked as read-only using a decorator are editable if you place your cursor position just before the first character of the variable and type.

## Expected Behavior
Should not be able to edit read-only variables.

## Current Behavior
Can insert characters before the first character of the variable.

## Steps to Reproduce
1. Add a signature block
2. Position cursor before the first character of the variable
3. Type

![image](https://user-images.githubusercontent.com/623108/94990662-d0f59500-0575-11eb-8266-8e5db08b9417.png)

### Desktop
 - OS: Mac OS
 - Browser: Safari
 - Version: Latest

**Comment by sahalsaad** [comment]
Hi @dselman ,

Sorry to ask. How to add signature block?

**Comment by dselman** [comment]
Open the Storybook, select Knobs and then select the Signature Block template in the drop down.

**Comment by K-Kumar-01** [comment]
> # Bug Report 
> Variables that are marked as read-only using a decorator are editable if you place your cursor position just before the first character of the variable and type.
> 
> ## Expected Behavior
> Should not be able to edit read-only variables.
> 
> ## Current Behavior
> Can insert characters before the first character of the variable.
> 
> ## Possible Solution
> ## Steps to Reproduce
> 1. Add a signature block
> 2. Position cursor before the first character of the variable
> 3. Type
> 
> ![image](https://user-images.githubusercontent.com/623108/94990662-d0f59500-0575-11eb-8266-8e5db08b9417.png)
> 
> ## Context (Environment)
> ### Desktop
> * OS: Mac OS
> * Browser: Safari
> * Version: Latest
> 
> ## Detailed Description
> ## Possible Implementation

I am not able to reproduce the issue.
I have inserted the signature block template and then made read only.(The text is not locked in my case).
After that the cursor arrow is not changing when hovering on field or before the first character.

**Comment by irmerk** [comment]
@K-Kumar-01 The issue is that the user shouldn't need to mark this as read only. Notice if you don't mark it as read only, you can type at the beginning of a signature field but not in the middle of the field.

**Comment by sanketshevkar** [comment]
Can I work on this bug?

**Comment by sanketshevkar** [comment]
I was able to reproduce this bug. Had a question, what is the difference between Variable in the Signature Block Template and Acceptance of Delivery Template?

**Comment by irmerk** [comment]
@dselman I think you'd be able to answer this best.

**Comment by sanketshevkar** [comment]
I have solved this bug, but I'm facing an issue with the build. Sometimes the text inside the clause is not editable whereas the variables are. Sometimes the build works just fine. I forked the project again but it's showing the same behavior. Also during the build, I'm getting a warning.
![eror](https://user-images.githubusercontent.com/43399522/107385018-859d5580-6b18-11eb-97e6-d99e3acb695c.png)
